tests/frontend/source-info/plugins/extradata.py

```diff
@@ -0,0 +1,44 @@
+from buildstream import Source
+
+
+class ExtraData(Source):
+    BST_MIN_VERSION = "2.0"
+
+    def configure(self, node):
+        pass
+
+    def preflight(self):
+        pass
+
+    def get_unique_key(self):
+        return {}
+
+    def load_ref(self, node):
+        pass
+
+    def get_ref(self):
+        return {}
+
+    def set_ref(self, ref, node):
+        pass
+
+    def is_cached(self):
+        return False
+
+    def collect_source_info(self):
+        return [
+            self.create_source_info(
+                "http://ponyfarm.com/ponies",
+                "pony-ride",
+                "pony-age",
+                "1234567",
+                version_guess="12",
+                extra_data={"pony": "green"},
+            )
+        ]
+
+
+# Plugin entry point
+def setup():
+
```

Comment:
nit: drop this newline? (here and in the other sources)

Reply:
Sure, for absolutely no reason at all, except that you like it, I have updated that :)